sdk/data/azcosmos/cosmos_container.go

```diff
@@ -138,9 +138,8 @@ func (c *ContainerClient) Delete(
 	}
 
 	operationContext := pipelineRequestOptions{
-		resourceType:     resourceTypeCollection,
-		resourceAddress:  c.link,
-		isWriteOperation: true,
+		resourceType:    resourceTypeCollection,
+		resourceAddress: c.link,
```

Comment:
A delete is a Write operation. And we have tests that use this. The Write operation is used to define routing (write operations can only be sent to the primary region)

Please verify with the service team the origin of the HTTP 500.

Reply:
isWriteOperation does not modify the request that is sent, it is used to know which region to route it to.

https://github.com/Azure/azure-sdk-for-go/blob/2532a53cfb2fabda9cbb1f04d209d95e84d18cb8/sdk/data/azcosmos/cosmos_location_cache.go#L116